src/waveform/renderers/allshader/waveformrendererstem.cpp

```diff
@@ -46,12 +50,43 @@ bool WaveformRendererStem::init() {
     for (int stemIdx = 0; stemIdx < mixxx::kMaxSupportedStems; stemIdx++) {
         QString stemGroup = EngineDeck::getGroupForStem(m_waveformRenderer->getGroup(), stemIdx);
         m_pStemGain.emplace_back(
-                std::make_unique<PollingControlProxy>(stemGroup,
+                std::make_unique<ControlProxy>(stemGroup,
                         QStringLiteral("volume")));
         m_pStemMute.emplace_back(
-                std::make_unique<PollingControlProxy>(stemGroup,
+                std::make_unique<ControlProxy>(stemGroup,
                         QStringLiteral("mute")));
+        auto bringToForeground = [this, stemIdx](double) {
+            if (!m_reorderOnChange) {
+                return;
+            }
+            m_stackOrder.removeAll(stemIdx);
+            m_stackOrder.append(stemIdx);
+        };
+        m_pStemGain.back()->connectValueChanged(this, bringToForeground);
+        m_pStemMute.back()->connectValueChanged(this, bringToForeground);
     }
+
+    m_stackOrder.resize(mixxx::kMaxSupportedStems);
+    std::iota(m_stackOrder.begin(), m_stackOrder.end(), 0);
+
+#ifndef __SCENEGRAPH__
```

Comment:
why is this disabled on scenegraph?

Reply:
Because on Scenegraph/Mixxx 3.0, the `WaveformWidgetFactory` isn't used and should eventually go away. Instead we will solely rely on property binding. If you want to see how that will look like, you might want to look [here](https://github.com/mixxxdj/mixxx/pull/14887/files#diff-e8d76e827e5ecce82c5bf5f4682f86f80054c094b9ff8e9a216094af2ca169a4R718) (obliviously, that
case is slightly different as this is the one needed for setting preview, but you get the gist of it)